Linkify extension doesn't match hosts that start with single-letter domain label

Google's **g.page** hostname is not being matched by the Linkify extension.

Example rendering by Github in this issue editor:  https://g.page/foo

Originally [reported by a Hugo user](https://github.com/gohugoio/hugo/issues/8143), but this is an issue with goldmark.

### Questionaire:

- [x] goldmark is fully compliant with the CommonMark. Before submitting issue, you **must** read [CommonMark spec](https://spec.commonmark.org/0.29/) and confirm your output is different from [CommonMark online demo](https://spec.commonmark.org/dingus/).
    - [x] **Extensions(Autolink without `<` `>`, Table, etc) are not part of CommonMark spec.** You should confirm your output is different from other official renderers correspond with an extension.
- [x] **goldmark is not dedicated for Hugo**. If you are Hugo user and your issue was raised by your experience in Hugo, **you should consider create issue at Hugo repository at first** .

**1. What version of goldmark are you using? :**

master (c4b3054802054bacb6445f3af7ce2d0fbbaaea1f)

**2. What version of Go are you using? :**

go1.16rc1
 
**3. What operating system and processor architecture are you using? :**

linux/amd64

**4. What did you do? :** 

```diff
diff --git a/extension/_test/linkify.txt b/extension/_test/linkify.txt
index d619a8b..134f52b 100644
--- a/extension/_test/linkify.txt
+++ b/extension/_test/linkify.txt
@@ -169,3 +169,11 @@ http://server.intranet.acme.com:1313
 //- - - - - - - - -//
 <p><a href="http://server.intranet.acme.com:1313">http://server.intranet.acme.com:1313</a></p>
 //= = = = = = = = = = = = = = = = = = = = = = = =//
+
+
+17
+//- - - - - - - - -//
+https://g.page/foo
+//- - - - - - - - -//
+<p><a href="https://g.page/foo">https://g.page/foo</a></p>
+//= = = = = = = = = = = = = = = = = = = = = = = =//
```
**5. What did you expect to see? :**

passing test

**6. What did you see instead? :**

```
--- FAIL: TestLinkify (0.00s)
    testutil.go:176: ============= case 17 ================
        Markdown:
        -----------
        https://g.page/foo

        Expected:
        ----------
        <p><a href="https://g.page/foo">https://g.page/foo</a></p>

        Actual
        ---------
        <p>https://g.page/foo</p>
```

**7. Did you confirm your output is different from [CommonMark online demo](https://spec.commonmark.org/dingus/) or other official renderer correspond with an extension?:**

Linkify extension not part of CommonMark spec.

--- a/extension/_test/linkify.txt
+++ b/extension/_test/linkify.txt
@@ -169,3 +169,11 @@ http://server.intranet.acme.com:1313
 //- - - - - - - - -//
 <p><a href="http://server.intranet.acme.com:1313">http://server.intranet.acme.com:1313</a></p>
 //= = = = = = = = = = = = = = = = = = = = = = = =//
+
+
+17
+//- - - - - - - - -//
+https://g.page/foo
+//- - - - - - - - -//
+<p><a href="https://g.page/foo">https://g.page/foo</a></p>
+//= = = = = = = = = = = = = = = = = = = = = = = =//

--- a/extension/linkify.go
+++ b/extension/linkify.go
@@ -11,9 +11,9 @@ import (
 	"github.com/yuin/goldmark/util"
 )
 
-var wwwURLRegxp = regexp.MustCompile(`^www\.[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]+(?:[/#?][-a-zA-Z0-9@:%_\+.~#!?&/=\(\);,'">\^{}\[\]` + "`" + `]*)?`)
+var wwwURLRegxp = regexp.MustCompile(`^www\.[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-z]+(?:[/#?][-a-zA-Z0-9@:%_\+.~#!?&/=\(\);,'">\^{}\[\]` + "`" + `]*)?`)
 
-var urlRegexp = regexp.MustCompile(`^(?:http|https|ftp)://[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]+(?::\d+)?(?:[/#?][-a-zA-Z0-9@:%_+.~#$!?&/=\(\);,'">\^{}\[\]` + "`" + `]*)?`)
+var urlRegexp = regexp.MustCompile(`^(?:http|https|ftp)://[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-z]+(?::\d+)?(?:[/#?][-a-zA-Z0-9@:%_+.~#$!?&/=\(\);,'">\^{}\[\]` + "`" + `]*)?`)
 
 // An LinkifyConfig struct is a data structure that holds configuration of the
 // Linkify extension.
@@ -24,10 +24,12 @@ type LinkifyConfig struct {
 	EmailRegexp      *regexp.Regexp
 }
 
-const optLinkifyAllowedProtocols parser.OptionName = "LinkifyAllowedProtocols"
-const optLinkifyURLRegexp parser.OptionName = "LinkifyURLRegexp"
-const optLinkifyWWWRegexp parser.OptionName = "LinkifyWWWRegexp"
-const optLinkifyEmailRegexp parser.OptionName = "LinkifyEmailRegexp"
+const (
+	optLinkifyAllowedProtocols parser.OptionName = "LinkifyAllowedProtocols"
+	optLinkifyURLRegexp        parser.OptionName = "LinkifyURLRegexp"
+	optLinkifyWWWRegexp        parser.OptionName = "LinkifyWWWRegexp"
+	optLinkifyEmailRegexp      parser.OptionName = "LinkifyEmailRegexp"
+)
 
 // SetOption implements SetOptioner.
 func (c *LinkifyConfig) SetOption(name parser.OptionName, value interface{}) {
@@ -156,10 +158,12 @@ func (s *linkifyParser) Trigger() []byte {
 	return []byte{' ', '*', '_', '~', '('}
 }
 
-var protoHTTP = []byte("http:")
-var protoHTTPS = []byte("https:")
-var protoFTP = []byte("ftp:")
-var domainWWW = []byte("www.")
+var (
+	protoHTTP  = []byte("http:")
+	protoHTTPS = []byte("https:")
+	protoFTP   = []byte("ftp:")
+	domainWWW  = []byte("www.")
+)
 
 func (s *linkifyParser) Parse(parent ast.Node, block text.Reader, pc parser.Context) ast.Node {
 	if pc.IsInLinkLabel() {

